### ISSUE TITLE
clientconfig.NewServiceClient should take a RegionName

### ISSUE THREAD

**Issue by rickard-von-essen** [gophercloud/utils#48]
It should be possible to pass in `REGION_NAME` to [`clientconfig.NewServiceClient`](https://github.com/gophercloud/utils/blob/master/openstack/clientconfig/requests.go#L432). 

I suggest adding an argument:

```golang
// NewServiceClient is a convenience function to get a new service client.
func NewServiceClient(service, region string, opts *ClientOpts) (*gophercloud.ServiceClient, error) {
```

And just wrap the [region part](https://github.com/gophercloud/utils/blob/master/openstack/clientconfig/requests.go#L474-L482):

```golang
	// Determine the region to use.
	if region == "" {
	        if v := cloud.RegionName; v != "" {
		        region = cloud.RegionName
	        }

	        if v := os.Getenv(envPrefix + "REGION_NAME"); v != "" {
		         region = v
	        }
        }
```

**Comment by rickard-von-essen** [comment]
I don't have a strong opinion, but I think I would prefer `ClientOpts` since it's not regarding auth.

**Comment by jtopjian** [comment]
@rickard-von-essen Yes, good point.

I've opened #49 to add this. Let me know if this looks good to you :)

**Comment by jtopjian** [comment]
Implemented in #49
